### DISCUSSION TITLE
How can I disable the automatic date formatting globally?

### DISCUSSION THREAD

**Discussion by jwhetzel-gpc** [saulpw/visidata#2012] [Q&A]
Hello!  I'm running into an issue where my date and datetime fields coming out of sqlite are getting formatted as dates only, and I need to disable this globally.  I saw in issue [#1572 ](https://github.com/saulpw/visidata/issues/1572) that the global fix is to set the desired format in `options.disp_date_fmt` in `~/.visidatarc`.  Unfortunately that won't work for me because what I really need to be able to to do is see the raw values in those columns so I can easily see which ones are date-only, which are timestamps in the format of `YYYY-mm-dd HH:MM:SS` and which are formatted with a "T" in between the date and time.  (`YYYY-mm-ddTHH:MM:SS`)  This way I can know what I'm dealing with on the query side.  Thanks!

**Comment by ajkerrigan** [comment]
Does using `type-any` (`z~`) or `type-string` (`~`) on that column help? Since the issue seems to be that you're losing context during the conversion to a Python datetime object, using the any/string types should avoid converting to a datetime object in the first place.

**Reply by jwhetzel-gpc** [reply]
They do work, but I would like to avoid the conversion globally every time I start visidata rather than having to remember convert them back to strings each time I look at a date or timestamp field.

**Reply by saulpw** [reply]
What sheet is this, that has a column automatically typed `date`?  If it's a static sheet and column named e.g. `date_column`, you can now (in develop) add this to your .visidatarc:

`Sheet.knownCols.date_column.type = anytype`

and it will automatically type all columns named `date_column` as the anytype.

**Reply by jwhetzel-gpc** [reply]
Thanks for that, Saul.  That will probably work for what I need if I run into it again in the future.  With my original question being so long ago, I unfortunately cannot seem to find where I was running into that problem previously.  I use Visidata for a lot of one-off stuff, so it's not something that I notice happening often.  If I catch it again in the future, I'll try to include some sample data.
